feat(components): ProgressBar can have a size

whereyoubinns opened the pull request:
<!--
  Atlantis uses Conventional Commits to track versions.
  Pull request titles should follow the following format.

  <TYPE>(<optional SCOPE>): <conditionally BREAKING CHANGE:> <description>

  eg.
    fix(SCOPE): stop graphite breaking when too much pressure applied — Patch Release
    feat(SCOPE): add 'graphiteWidth' option — (Minor) Feature Release
    feat(SCOPE): BREAKING CHANGE: remove graphiteWidth option — (Major) Breaking Release

  TYPE should consist of:
    - fix: a commit of the type fix patches a bug in your codebase
    - feat: a commit of the type feat introduces a new feature to the codebase
    - docs: documentation only changes
    - build: improvements to the build system
    - refactor: a change that neither fixes a bug nor introduces a feature
    - chore: other changes that don't modify src or test files

  SCOPE should be one of:
    - components
    - design
    - eslint
    - generators
    - hooks
    - stylelint


  If your pull request introduces a breaking change please append `BREAKING CHANGE:` following type / scope.

  Further Reading:
    - https://www.conventionalcommits.org
    - https://github.com/commitizen/conventional-commit-types/blob/master/index.json
-->

## Motivations

The progress bar was only one size that was too large when using it in smaller spaces. 

## Changes

<!-- https://keepachangelog.com/en/1.0.0/ -->

### Added

ProgressBar can have a small size, which is half the size of the default ProgressBar. 

### Changed

- <!-- changes in existing functionality -->

### Deprecated

- <!-- soon-to-be removed features -->

### Removed

- <!-- now removed features -->

### Fixed

- <!-- for any bug fixes -->

### Security

- <!-- in case of vulnerabilities -->

## Testing

<!-- How to test your changes. -->

---

[In Atlantis we use Github's built in pull request reviews](https://help.github.com/en/articles/about-pull-request-reviews).

![Random photo of Atlantis](https://loremflickr.com/672/400/atlantis)

JeffMarvin commented on packages/components/src/ProgressBar/ProgressBar.tsx:
```diff
@@ -12,12 +13,24 @@ interface ProgressBarProps {
+export function ProgressBar({
+  currentStep,
+  totalSteps,
+  size,
```
size 
```suggestion
  size = "base",
```

JeffMarvin commented on packages/components/src/ProgressBar/ProgressBar.tsx:
```diff
@@ -12,12 +13,24 @@ interface ProgressBarProps {
+  /**
+   * Changes the size to small
+   */
+  readonly size?: keyof typeof sizes;
```
In other places in Atlantis, we've just used strings. You could do:
```suggestion
  readonly size?: "small" | "base";
```

seanhealy replied:
What @whereyoubinns has here is actually probably better for this case. It is saying that size can be any of the "classes" exported by `sizes.css`. So, if we just had the base size there we're golden!

JeffMarvin commented on packages/components/src/ProgressBar/ProgressBar.tsx:
```diff
@@ -12,12 +13,24 @@ interface ProgressBarProps {
   readonly totalSteps: number;
+
+  /**
+   * Changes the size to small
```
Suggestion:
```suggestion
   * Set the size of the progress bar (optional)
```

seanhealy replied:
`(optional)` will be inferred by the docs viewer.

JeffMarvin commented on packages/components/src/ProgressBar/ProgressBar.tsx:
```diff
@@ -12,12 +13,24 @@ interface ProgressBarProps {
-  const progressBarContainerClassName = classnames(styles.container);
+  const progressBarContainerClassName = classnames(
+    styles.container,
+    size && sizes[size],
```
You'd have to adjust this, depending on what other suggestions from the PR you take. :)

seanhealy commented on packages/components/src/ProgressBar/Sizes.css:
```diff
@@ -0,0 +1,3 @@
+.small {
+  --progress-bar--height: 0.5rem;
```
Would be better to use the spacing units here.

https://atlantis.getjobber.com/space

While you're in here do you want to double check that the "base" size is also using the spacing units? 😄

whereyoubinns commented:
Made some changes based on both your feedback! I kept the enumerable sizes on seans suggestion and updated some of the wording. 

I also realized our use case actually needs a progress bar with the "smaller" size, so I updated the spacing accordingly. 

@JeffMarvin / @seanhealy could you do another review please :)

seanhealy commented on packages/components/src/ProgressBar/Sizes.css:
```diff
@@ -0,0 +1,7 @@
+.smaller {
```
Do we anticipating having a `small` size in the future that would be between `base` and `smaller`? If not this might be better as just `small` since this is what a user of the component will need to specify when they are using it and specifying a size.

cc. @chris-at-jobber 

```suggestion
.small {
```

seanhealy commented on packages/components/src/ProgressBar/ProgressBar.css:
```diff
@@ -1,6 +1,5 @@
 :root {
   --progress-bar--borderRadius: 1rem;
```
We should update this to be a percentage I think so that the radius scales properly with the height.

whereyoubinns commented:
@seanhealy updated both of those :)
